### DISCUSSION TITLE
File size limit?

### DISCUSSION THREAD

**Discussion by cfus98** [audiamus/AaxAudioConverter#48] [Q&A] [answered]
Hi. I am converting a couple of very long books from .aax to .mp3 (eg. 1193MB/85hrs/32kbs and 1326M/48hrs/64kbs) and the conversions are failing. Any ideas why this is happening and do you have any suggestions to fix it? Should I just download my books in parts instead of a single file? Or, should I split the output into multiple files (shorter tracks)? Thanks for your help.

**Comment by audiamus** [comment, marked as answer]
I don't really know. I am not aware of a file size limit for MP3. 

To find out the cause of the error, run the converter with logging on (see the manual for instructions, chapter 4). If you get the "aborted" message, it could be an error detected by FFmpeg. In that case, log level should be set to 4, otherwise 3. Note: With level 4, the log files will become quite long. 

Shorter files often make sense, in several respects. Multi-part AAX downloads are less likely to fail. In the feedback I receive, corrupt files are the main reason for failure, and shorter downloads have a lower risk here. For the converted book, shorter MP3 files are easier to navigate. And there will be the playlist, to keep the files together. The playlist will always be created for a multi-part book or any conversion mode other than "single file". And, particularly for MP3 and a multi-core processor, conversion into multiple output files will be significantly faster, as the converter will be able to parallalize the work.

**Comment by cfus98** [comment]
Good Morning and Happy New Year. I redownloaded the audiobooks as parts instead of a single file. No problems to report. And, yes considerably faster due to parallalized work. Thanks for the tip!

**Comment by audiamus** [comment]
The cause for the conversion failure for very long books appears to be an FFmpeg memory allocation problem. Not sure why FFmpeg does not release memory while transcoding - in my understanding this a stream operation after all - but it eventually runs out of address space. The FFmpeg executable here is a 32 bit build.    

Currently, the AAX Audio Converter installation package is still bundled with the 32bit build of FFmpeg for the simple reason that that will run on both 32 and 64 bit systems. (AAX Audio Converter itself is an "Any CPU" build that will run as a 64bit process on 64bit Windows and as a 32bit process on 32bit Windows.)

From the next release onward AAX Audio Converter will ship also with a 64 bit build of FFmpeg and the installer will select which one to install, depending to the operating system.
